core/src/miner/mem_pool.rs

```diff
@@ -246,12 +245,19 @@ impl MemPool {
             let order = TransactionOrder::for_transaction(&item, client_account.seq);
             let order_with_tag = TransactionOrderWithTag::new(order, QueueTag::New);
 
+            if !if_recovering {
+                backup::backup_item(&mut batch, hash, &item);
```

Comment:
Not every item is inserted in here. See the line 329: `self.enforce_limit();`.
I think you need to handle items in `enforce.limit()` too.

Reply:
Thanks, I applied backup also in `enforce_limit` function